js/kontent.js

```diff
@@ -0,0 +1,63 @@
+const projectId = "99b0b3b0-4838-0051-3d57-8af72f55e8a0";
+const apiBlogItems = `https://deliver.kontent.ai/${projectId}/items`;
+const apiBlogItem = `https://deliver.kontent.ai/${projectId}/items/`;
+const apiTaxonomies = `https://deliver.kontent.ai/${projectId}/taxonomies/`;
+const pageSize = 10;
+
+class URLBuilder {
+    constructor(baseURL) {
+        this.singleEndpoint = null;
+        this.includeTotalCount = true;
+        this.limit = null;
+        this.orderByElementProp = null;
+        this.blogTagsContains = null;
+        this.baseURL = new URL(baseURL + (this.singleEndpoint || ""));
+    }
+
+    build() {
+        const working = new URL(this.baseURL + (this.singleEndpoint || ""));
+        const searchParams = working.searchParams;
+        if (this.includeTotalCount) searchParams.set("includeTotalCount", this.includeTotalCount);
+        if (this.limit) searchParams.set("limit", this.limit);
+        if (this.orderByElementProp) searchParams.set("order", `elements.${this.orderByElementProp}`);
+        if (this.blogTagsContains) searchParams.set("elements.blog_tags[contains]", this.blogTagsContains);
+        return working;
+    }
+}
+const itemsFetchKontentApi = new URLBuilder(apiBlogItems);
+const itemFetchKontentApi = new URLBuilder(apiBlogItem);
+const taxonomiesFetchKontentApi = new URLBuilder(apiTaxonomies);
+
+
+export function buildBlogItemsUrl(tag) {
+    itemsFetchKontentApi.includeTotalCount = true;
```

Comment:
could this initialization be done when initializing `itemsFetchKontentApi` ? Is `itemsFetchKontentApi` ever used without
```javascript
includeTotalCount = true;
limit = pageSize; 
orderByElementProp = "post_date[desc]";
```

?

Reply:
We can set the pageSize const as the default, total count default is already true, and order by can vary...